include/aikido/perception/PoseEstimatorModule.hpp

```diff
@@ -40,19 +43,25 @@ class PoseEstimatorModule : public PerceptionModule
   /// pose is transformed
   PoseEstimatorModule(
       ros::NodeHandle nodeHandle,
-      std::string markerTopic,
-      std::shared_ptr<ObjectDatabase> configData,
+      const std::string& markerTopic,
+      std::shared_ptr<AssetDatabase> assetData,
       std::shared_ptr<aikido::io::CatkinResourceRetriever> resourceRetriever,
-      std::string referenceFrameId,
+      const std::string& referenceFrameId,
       dart::dynamics::Frame* referenceLink);
 
   virtual ~PoseEstimatorModule() = default;
 
   // Documentation inherited
+  /// Looks for the following information sent via ROS (see \c DetectedObject):
+  /// visualization_msgs/Marker message like the following:
+  /// Marker.header.frame_id -> detectionFrameID
+  /// Marker.ns + "_" + Marker.id -> objUid (identity in DART world)
+  /// Marker.ns -> objAssetDBKey (determines visual asset, see AssetDatabase)
   bool detectObjects(
       const aikido::planner::WorldPtr& env,
-      ros::Duration timeout = ros::Duration(0.0),
-      ros::Time timestamp = ros::Time(0.0)) override;
+      ros::Duration timeout = ros::Duration(),
+      ros::Time timestamp = ros::Time(0.0),
+      std::vector<DetectedObject>* detectedObjects = nullptr) override;
```

Comment:
It seems like the default params are specified only here and not in `PerceptionModule`? And `PerceptionModule`'s document seems to suggest otherwise. Could you move these default values to `PerceptionModule`?

Reply:
Defaults need to be present in the subclass that is actually being called.